### PR TITLE
Supplied functions

### PR DESCRIPTION
**Pull request by boxdot**

### CODE REVIEW

**Comment by boxdot** [comment]
Resolves #12.

File: src/interpreter.rs [hunk 1]
```diff
@@ -423,15 +444,92 @@ impl<'a> Interpreter<'a> {
         Ok(acc)
     }
 
-    fn evaluate_primary<W: Write>(&self, primary: &Primary, stderr: &mut W) -> Result<f64, Error> {
+    fn evaluate_primary<W: Write>(
+        &mut self,
```
Comment (jeschkies) [inline src/interpreter.rs]: Is `self` mutable because of the random number generator?

File: tests/suite/P130.ok [hunk 2]
```diff
@@ -10,26 +10,26 @@ RND FUNCTION.
...
- 12            .628871 
- 13            .364784 
- 14            .513401 
- 15            .95223 
- 16            .916195 
- 17            .635712 
- 18            .717297 
- 19            .141603 
- 20            .606969 
+ 1             .168183
```
Comment (jeschkies) [inline tests/suite/P130.ok]: This is odd. Did we upload the wrong test program?

File: src/interpreter.rs (hunk 1 above)
Comment (boxdot) [reply src/interpreter.rs]: Yeah, exactly. I thought about using inferior mutability for that later. Actually, the idea would be to create a devices struct which would contain all devices like rng, stdout, stdin, screen, etc... and only this one will need to be mut.

File: tests/suite/P130.ok (hunk 2 above)
Comment (boxdot) [reply tests/suite/P130.ok]: No, but the numbers are random and there is no spec what is the seed and how exactly to generate the numbers (luckely enough).

**Pull request merged**